io.openems.edge.timedata.mssql/src/io/openems/edge/timedata/mssql/TimedataMssqlImpl.java

```diff
@@ -44,44 +63,50 @@ public TimedataMssqlImpl() {
 				TimedataMssql.ChannelId.values() //
 		);
 	}
-
+	
+	@Reference
+	protected ComponentManager componentManager;
+	
 	@Activate
 	void activate(ComponentContext context, Config config) {
 		super.activate(context, config.id(), config.alias(), config.enabled());
+		this.mssqlConnector = new MssqlConnector(config.server(), config.dbname(), config.port(), 
+				config.username(), config.password(), config.isPasswordProtected(), config.isReadOnly());
+		
 		this.config = config;
```

Comment:
@sfeilmeier

Hi Stefan,  
At this point of our discussion, I would also need your suggestion, with PreparedStatements. I have planned that I write down the Insert Strings for Prepared statements in another file, and simple call them and activate the prepared statements too during activation of the connections. As per my understanding, I would be compling the statements only once during the whole life cycle of the code, i.e. during activation, and then every data insert would be carried out using the PreparedStatements initiated here. 

Do you think it is a good idea, or could it be somehow made more simpler?

Thanks and regards
Laksh

Reply:
Have you seen the Postgres implementation in the Metadata.Odoo bundle? There I am also using PreparedStatements: https://github.com/OpenEMS/openems/blob/develop/io.openems.backend.metadata.odoo/src/io/openems/backend/metadata/odoo/postgres/task/InsertEdgeConfigUpdate.java

If you only have one long-living connection, you should be able to proceed as you described.